lib/matplotlib/__init__.py

```diff
@@ -732,6 +732,11 @@ def __setitem__(self, key, val):
                 if val is rcsetup._auto_backend_sentinel:
                     if 'backend' in self:
                         return
+            elif key == 'boxplot.vertical' and not val:
```

Comment:
I'm not clear how the internals of of `rcParams` work together but, in a test this does not raise a warning when the parameter is set in matplotlibrc or when set via `plt.rc_context`. Maybe it's simpler to put the warning to where the value is read in `bxp`?

Either way, I don't care too much about how this is implemented, but I would like to see a test that ensures the warning on the rcParam is issued.

Reply:
The warnings are suppressed for some reason when it's set that way. I just put it in `bxp`